lib/1.4/utility.dml

```diff
@@ -1371,6 +1371,12 @@ template map_target is (connect, _qname) {
                 return Sim_PE_IO_Not_Taken;
             }
         }
-        return SIM_issue_transaction(map_target, t, addr);
+        local exception_type_t exc = SIM_issue_transaction(map_target, t, addr);
+        local bool fail = exc != Sim_PE_No_Exception;
+        log info, (fail ? 2 : 4) : "%s%s %d bytes @ 0x%x in %s",
+            fail ? "failed to " : "",
+            SIM_transaction_is_read(t) ? "read" : "write",
```

Comment:
Inconsistent grammar: `{failed to read, failed to writre, read} N bytes` all qualify as past tense, whereas `write N bytes` does not.

Reply:
Got it, I'll change `failed => fail` and add an exclamation mark, making the grammar consistently imperative tense ;-)

Reply:
It seems that this change didn't make it into your latest push.

I would advise against an exclamation mark; we don't use punctuation in short messages and an exclamation mark makes the tone less neutral. (Also I'm not *against* past tense when you describe something in the past; my problem was with the inconsistency. I don't know if there's a recommendation on past vs imperative here)

Reply:
Sorry, the comment was a joke. My latest push changes the grammar to consistently past tense.
> read 8 bytes @ 0x0 in phys_mem
> wrote 8 bytes @ 0x0 in phys_mem
> failed to read 8 bytes @ 0x0 in phys_mem
> failed to write 8 bytes @ 0x0 in phys_mem

```
        log info, (fail ? 2 : 4) : "%s%s %d bytes @ 0x%x in %s",
            fail ? "failed to " : "",
            SIM_transaction_is_read(t) ? "read" : fail ? "write" : "wrote",
            SIM_transaction_size(t), addr, SIM_object_name(obj);
        return exc;
```